src/main/java/net/smartcosmos/dao/metadata/MetadataDao.java

```diff
@@ -14,55 +15,84 @@
 public interface MetadataDao {
 
     /**
-     * Upserts a list of metadata entities associated to a reference entity in the realm of a given account.
+     * Inserts a list of metadata entities associated to a reference entity in the realm of a given tenant.
      *
-     * @param accountUrn the account URN
-     * @param upsertMetadataCollection the collection containing the metadata entries to upsert
-     * @return an {@link MetadataResponse} list for the upserted metadata
-     * @throws ConstraintViolationException if the {@link MetadataUpsert} violates constraints enforced by the persistence service
+     * @param tenantId the tenant URN
+     * @param createMetadataCollection the collection containing the metadata entries to insert
+     * @return an {@link MetadataResponse} list for the inserted metadata
+     * @throws ConstraintViolationException if the {@link MetadataCreate} violates constraints enforced by the persistence service
      */
-    List<MetadataResponse> upsert(String accountUrn, Collection<MetadataUpsert> upsertMetadataCollection) throws ConstraintViolationException;
+    List<MetadataResponse> create(String tenantId, Collection<MetadataCreate> createMetadataCollection)
+            throws ConstraintViolationException;
 
     /**
-     * Deletes an existing metadata key that is associated to an entity (identified by its reference URN key).
+     * Updates a list of metadata entities associated to a reference entity in the realm of a given tenant.
      *
-     * @param accountUrn the account URN
-     * @param entityReferenceType the reference type of the associated entity
-     * @param referenceUrn the URN of the associated entity
-     * @param key the key of the metadata entity
+     * @param tenantId the tenant URN
+     * @param updateMetadataCollection the collection containing the metadata entries to update
+     * @return an {@link MetadataResponse} list for the updated metadata
+     * @throws ConstraintViolationException if the {@link MetadataUpdate} violates constraints enforced by the persistence service
+     */
+    List<MetadataResponse> update(String tenantId, Collection<MetadataUpdate> updateMetadataCollection)
```

Comment:
For now, we don't plan to support updating multiple entries at once. Thus, there is no need for a collection.

Moreover, since almost everything is in the request URL, there is no need for a `MetadataUpdate` object. The method signature could just be `update(String tenantId, String ownerType, String ownerId, String metadataKey, Object metadataValue)`